src/core/net/ip6.cpp

```diff
@@ -1291,8 +1289,18 @@ Error Ip6::HandleDatagram(Message &aMessage, MessageOrigin aOrigin, const void *
             VerifyOrExit(isAllowedType, error = kErrorDrop);
         }
 
+        if (aMessage.GetOrigin() == Message::kOriginHostUntrusted && nextHeader == kProtoUdp)
```

Comment:
If we are dropping a message due to untrusted source, should we provide some more visible (e.g. NOTE level) log message?

Reply:
Added a logging line for this.